rep-0149.rst

```diff
@@ -512,6 +512,25 @@ Commonly used license strings:
  - MIT
  - Mozilla Public License Version 1.1
 
+Attributes
+''''''''''
+
+  .. raw:: html
+
+    <font color="blue">
+
+ ``file="FILE"`` *(optional)*
+
+  A relative path from ``package.xml`` pointing the full license text.
+
+  Some open source licenses require provision of the license text when a user
```

Comment:
I'm afraid the answer might be probably "no", but do we not want to provide some example licenses that require license text reference? If that'd be available I'd really appreciate that...

Reply:
I think it's OK to include examples.

How about
```
Some open source licenses such as BSD, MIT and LGPL require ...
```
is this enough?

Reply:
+1 for mentioning some commonly used licenses.

If adding some references like this would be ok, then I'd also suggest to include links to the official statement at the time of REP's publishment, such as:

Apache 2.0 https://www.apache.org/licenses/LICENSE-2.0
BSD https://opensource.org/licenses/BSD-3-Clause
L-GPL v3 https://www.gnu.org/licenses/lgpl-3.0.en.html
MIT https://opensource.org/licenses/MIT

Reply:
I think the question is which licenses do **require** the file with the license content to be present. I am not sure if all four licenses mentioned actually do...

Reply:
I think at least those four licenses require.

1. Apache 2.0
    > 4.a 
    > You must give any other recipients of the Work or Derivative Works a copy of this License; and

2. BSD
    >2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.

3. L-GPL v3
    >4-b)
    >b) Accompany the Combined Work with a copy of the GNU GPL and this license document.
4. MIT
    >The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

Reply:
Hi,
I found that most open source licenses require inclusion of license and copyright notice.
https://choosealicense.com/appendix/#include-copyright

I'm wondering if it is still better to include some specific examples. Rather than that, "Most open source licenses require ... [18]_. [18]https://choosealicense.com/appendix" is better?

Reply:
Regardless of a specific license's requirements, the new tag should be allowed for any license, particularly unfamiliar ones.

Reply:
> I found that most open source licenses require inclusion of license and copyright notice.
https://choosealicense.com/appendix/#include-copyright
>
> I'm wondering if it is still better to include some specific examples. Rather than that, "Most open source licenses require ... [18]_. [18]https://choosealicense.com/appendix" is better?

+1 for referring both kinds of links maybe in `Reference` section.
- Referencing to the official links like those I referred to in https://github.com/ros-infrastructure/rep/pull/157#discussion_r168868982 might be suitable for a formal document like REP?
- That said, the link on choosealicense.com looks very useful and I'd appreciate if a link to an external resource like that is readily available.

> Regardless of a specific license's requirements, the new tag should be allowed for any license, particularly unfamiliar ones.

I think it's a good idea to add a note like that.

Reply:
> Regardless of a specific license's requirements, the new tag should be allowed for any license, particularly unfamiliar ones.

That's right. The file attribution is needed not only to meet the license's requirements but also to clarify the unfamiliar licenses. I will add it.

> +1 for referring both kinds of links maybe in Reference section.

I will try.